File: apps/sepuling-java/src/main/java/smecalculus/rolevod/storage/SepulkaStateMapper.java
package smecalculus.rolevod.storage;

import org.mapstruct.Mapper;
import smecalculus.rolevod.core.SepulkaStateDm;
import smecalculus.rolevod.mapping.EdgeMapper;

@Mapper
public interface SepulkaStateMapper extends EdgeMapper {
    SepulkaStateEm.AggregateRoot toEdge(SepulkaStateDm.AggregateRoot state);

    SepulkaStateDm.AggregateRoot toDomain(SepulkaStateEm.AggregateRoot state);

    SepulkaStateEm.Touch toEdge(SepulkaStateDm.Touch state);

    SepulkaStateDm.Existence toDomain(SepulkaStateEm.Existence state);

    SepulkaStateEm.Preview toEdge(SepulkaStateDm.Preview state);

    SepulkaStateDm.Preview toDomain(SepulkaStateEm.Preview state);
}

File: apps/sepuling-kotlin/src/main/kotlin/smecalculus/rolevod/storage/SepulkaStateMapper.kt
package smecalculus.rolevod.storage

import org.mapstruct.Mapper
import smecalculus.rolevod.core.SepulkaStateDm
import smecalculus.rolevod.mapping.EdgeMapper

@Mapper
interface SepulkaStateMapper : EdgeMapper {
    fun toEdge(state: SepulkaStateDm.AggregateRoot): SepulkaStateEm.AggregateRoot

    fun toDomain(state: SepulkaStateEm.AggregateRoot): SepulkaStateDm.AggregateRoot

    fun toEdge(state: SepulkaStateDm.Touch): SepulkaStateEm.Touch

    fun toDomain(state: SepulkaStateEm.Existence): SepulkaStateDm.Existence

    fun toEdge(state: SepulkaStateDm.Preview): SepulkaStateEm.Preview

    fun toDomain(state: SepulkaStateEm.Preview): SepulkaStateDm.Preview
}
